### PR TITLE
Implement proper benchmarking

### LINKED ISSUE #14: Do proper benchmarking
Decide on a common dataset to do a proper benchmarking.

Ideally, it should be able to compare both inference speed and the vector quality, e.g., zero-shot labeling, image retrieval etc.

### PR DESCRIPTION
**Pull request by monatis**
Closes #14 

Run:
```shell
./bin/benchmark
```

to see how to use it. `images_dir` should have subdirectories each one which is named a class in the dataset. It calculates the accuracy at 1 and accuracy at 5.

We may consider implement proper multithreading to speed up the calculation later on.

- [ ] - [ ] Consider moving `get_dir_keyed_files` function to `clip.cpp`.

### CODE REVIEW

**Comment by monatis** [comment]
cc @Green-Sky

**Comment by Green-Sky** [comment]
```
CMake Error at CMakeLists.txt:218 (add_subdirectory):
  The source directory

    /home/green/workspace/clip.cpp/tests

  does not contain a CMakeLists.txt file.
```

**Comment by Green-Sky** [comment]
oh cool, you stared directory iteration without `std::filesystem`

**Comment by monatis** [comment]
I gitignored `tests/*.txt` previously because I was dumping some intermediate tensor output there for debugging. fixed now.

**Comment by monatis** [comment]
> you stared directory iteration without std::filesystem

we can move it to the main library if you want to use it in the image search example. And it may also be a direct use case for lib users. but I'm not sure if we should.

**Comment by monatis** [comment]
I'll also implement printing the output as a markdown table. maybe optionally dumping to stdout or to a regular file.

**Comment by Green-Sky** [comment]
> > you stared directory iteration without std::filesystem
> 
> we can move it to the main library if you want to use it in the image search example. And it may also be a direct use case for lib users. but I'm not sure if we should.

I would prefer something similar to llama.cpp/examples/common.cpp
so move it there, including cli parameter parsing..

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,2 @@
 build/
 .vscode
-models/*.bin
```
Comment (Green-Sky) [inline .gitignore:3]: why remove models/*.bin ?

Comment (monatis) [reply .gitignore:3]: don't remember the exact reason. probably it was a temporary change then I forgot about re-adding it. It should definitely be added again.

**Pull request merged**